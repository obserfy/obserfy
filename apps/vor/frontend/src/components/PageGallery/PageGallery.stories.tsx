import { Meta, Story } from "@storybook/react"
import PageGallery, { PageGalleryProps } from "./PageGallery"

export default {
  title: "Core/PageGallery",
  component: PageGallery,
  parameters: {
    componentSubtitle: "Just a simple PageGallery",
  },
} as Meta

const Template: Story<PageGalleryProps> = (args) => <PageGallery {...args} />

export const Default = Template.bind({})
Default.args = {}
