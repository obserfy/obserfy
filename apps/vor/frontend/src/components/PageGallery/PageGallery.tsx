import { keyframes } from "@emotion/react"
import { Trans } from "@lingui/macro"
import { ChangeEvent, FC, Fragment, useEffect, useState } from "react"
import { Box, Button, Flex, Image, Label } from "theme-ui"
import { getFirstName } from "../../domain/person"
import { VideoStatus } from "../../domain/video"
import { useUploadStudentVideo } from "../../hooks/api/schools/useUploadStudentVideo"
import useGetStudentImages, {
  StudentImage,
} from "../../hooks/api/students/useGetStudentImages"
import useGetVideos from "../../hooks/api/students/useGetVideos"
import usePostNewStudentImage from "../../hooks/api/students/usePostNewStudentImage"
import { useGetStudent } from "../../hooks/api/useGetStudent"
import useVisibilityState from "../../hooks/useVisibilityState"
import { ReactComponent as PlusIcon } from "../../icons/plus.svg"
import {
  STUDENT_IMAGE_URL,
  STUDENT_OVERVIEW_URL,
  STUDENTS_URL,
} from "../../routes"
import Dialog from "../Dialog/Dialog"
import Icon from "../Icon/Icon"
import { Link } from "../Link/Link"
import LoadingIndicator from "../LoadingIndicator/LoadingIndicator"
import Tab from "../Tab/Tab"
import TopBar, { breadCrumb } from "../TopBar/TopBar"
import TranslucentBar from "../TranslucentBar/TranslucentBar"
import Typography from "../Typography/Typography"
import VideoPlayerDialog from "../VideoPlayerDialog/VideoPlayerDialog"

export interface PageGalleryProps {
  studentId: string
}

enum View {
  IMAGES,
  VIDEOS,
}

const PageGallery: FC<PageGalleryProps> = ({ studentId }) => {
  const student = useGetStudent(studentId)
  const [selectedView, setSelectedView] = useState(View.IMAGES)

  return (
    <Box>
      <TranslucentBar boxSx={{ position: "sticky", top: -48, zIndex: 10 }}>
        <TopBar
          breadcrumbs={[
            breadCrumb("Students", STUDENTS_URL),
            breadCrumb(
              getFirstName(student.data),
              STUDENT_OVERVIEW_URL(studentId)
            ),
            breadCrumb("Gallery"),
          ]}
        />

        <Tab
          items={["Images", "Videos"]}
          selectedItemIdx={selectedView}
          onTabClick={(idx) => setSelectedView(idx)}
        />
      </TranslucentBar>

      {selectedView === View.IMAGES && <ImagesView studentId={studentId} />}
      {selectedView === View.VIDEOS && <VideosView studentId={studentId} />}
    </Box>
  )
}

const ImagesView: FC<{ studentId: string }> = ({ studentId }) => {
  const postNewStudentImage = usePostNewStudentImage(studentId)
  const images = useGetStudentImages(studentId)

  const handleImageUpload = async (event: ChangeEvent<HTMLInputElement>) => {
    const selectedImage = event.target.files?.[0]
    if (selectedImage) {
      try {
        await postNewStudentImage.mutateAsync(selectedImage)
      } catch (e) {
        Sentry.captureException(e)
      }
    }
  }

  return (
    <Box>
      <Flex>
        <Label
          px={3}
          pt={3}
          pb={[3, 2]}
          ml={[0, "auto"]}
          sx={{ width: ["100%", "auto"], display: "inline-block" }}
        >
          <input
            data-cy="upload-image"
            type="file"
            accept="image/*"
            style={{ display: "none" }}
            disabled={postNewStudentImage.isLoading}
            onChange={handleImageUpload}
          />
          <Button
            as="div"
            disabled={postNewStudentImage.isLoading}
            sx={{ width: ["100%", "auto"] }}
          >
            {postNewStudentImage.isLoading && <LoadingIndicator />}
            <Icon as={PlusIcon} mr={2} fill="onPrimary" />
            <Trans>Upload Photo</Trans>
          </Button>
        </Label>
      </Flex>

      <Flex px={[2, 2]} sx={{ width: "100%", flexWrap: "wrap" }}>
        {images.data?.map((image) => (
          <ImageItem key={image.id} studentId={studentId} image={image} />
        ))}
      </Flex>
    </Box>
  )
}

const fading = keyframes`
  0% {
    background-color: rgba(165, 165, 165, 0.05);
  }
  50% {
    background-color: rgba(165, 165, 165, 0.2);
  }
  100% {
    background-color: rgba(165, 165, 165, 0.05);
  }
`
// TODO: These are some ugly css, might be inconsistent on some devices
//  due to the calc and decimal points, revisit later.
const ImageItem: FC<{ studentId: string; image: StudentImage }> = ({
  studentId,
  image,
}) => (
  <Box
    data-cy="image"
    p={[1, 2]}
    sx={{ width: ["33.333%", "25%", "20%", "14.285%"] }}
  >
    <Link
      to={STUDENT_IMAGE_URL(studentId, image.id)}
      sx={{ animation: `1s ease-in-out 0s infinite ${fading}` }}
    >
      <Box pt="100%" sx={{ width: "100%", position: "relative" }}>
        <Image
          loading="lazy"
          src={image.thumbnailUrl}
          sx={{
            backgroundColor: "surface",
            position: "absolute",
            top: 0,
            bottom: 0,
            left: 0,
            right: 0,
            cursor: "pointer",
          }}
        />
      </Box>
    </Link>
  </Box>
)

const VideosView: FC<{ studentId: string }> = ({ studentId }) => {
  const [shouldPoll, setShouldPoll] = useState(false)
  const videos = useGetVideos(studentId, shouldPoll)
  const uploadVideo = useUploadStudentVideo(studentId)

  const handleVideoUpload = async (event: ChangeEvent<HTMLInputElement>) => {
    const selectedVideo = event.target.files?.[0]
    if (selectedVideo) {
      try {
        await uploadVideo.mutateAsync(selectedVideo)
      } catch (e) {
        Sentry.captureException(e)
      }
    }
  }

  useEffect(() => {
    const imageNotReady = videos.data?.find(
      (v) => v.status !== VideoStatus.READY
    )
    if (imageNotReady) {
      setShouldPoll(true)
    } else {
      setShouldPoll(false)
    }
  }, [videos.data])

  return (
    <Fragment>
      <div>
        <Flex>
          <Label
            px={3}
            pt={3}
            pb={[3, 2]}
            ml={[0, "auto"]}
            sx={{ width: ["100%", "auto"], display: "inline-block" }}
          >
            <input
              type="file"
              accept="video/*"
              style={{ display: "none" }}
              disabled={uploadVideo.isLoading}
              onChange={handleVideoUpload}
            />
            <Button
              as="div"
              disabled={uploadVideo.isLoading}
              sx={{ width: ["100%", "auto"] }}
            >
              {uploadVideo.isLoading ? (
                <Fragment>
                  <LoadingIndicator mr={2} />
                  <Trans>Uploading</Trans>
                </Fragment>
              ) : (
                <Fragment>
                  <Icon as={PlusIcon} mr={2} fill="onPrimary" />
                  <Trans>Upload Video</Trans>
                </Fragment>
              )}
            </Button>
          </Label>
        </Flex>

        <Flex px={[2, 2]} sx={{ width: "100%", flexWrap: "wrap" }}>
          {videos.data?.map((video) => (
            <VideoItem
              key={video.id}
              status={video.status}
              videoId={video.id}
              studentId={studentId}
              thumbnailUrl={video.thumbnailUrl}
              playbackUrl={video.playbackUrl}
              originalThumbnailUrl={video.originalThumbnailUrl}
              createdAt={video.createdAt}
            />
          ))}
        </Flex>
      </div>
      {uploadVideo.isLoading && (
        <Dialog>
          <Flex p={3} sx={{ alignItems: "center" }}>
            <Typography.H6 sx={{ alignText: "center" }}>
              Uploading video
            </Typography.H6>
            <LoadingIndicator ml="auto" size={30} color="primary" />
          </Flex>
        </Dialog>
      )}
    </Fragment>
  )
}

const VideoItem: FC<{
  studentId: string
  playbackUrl: string
  thumbnailUrl: string
  originalThumbnailUrl: string
  createdAt: string
  videoId: string
  status: string
}> = ({
  createdAt,
  thumbnailUrl,
  studentId,
  playbackUrl,
  originalThumbnailUrl,
  videoId,
  status,
}) => {
  const videoDialog = useVisibilityState()

  const isReady = status === VideoStatus.READY

  return (
    <Fragment>
      <Box p={[1, 2]} sx={{ width: ["33.333%", "25%", "20%", "14.285%"] }}>
        <Box
          sx={{
            display: "block",
            animation: `1s ease-in-out 0s infinite ${fading}`,
          }}
          onClick={() => {
            if (isReady) videoDialog.show()
          }}
        >
          <Box
            pt="100%"
            sx={{
              width: "100%",
              position: "relative",
              animation: `1s ease-in-out 0s infinite ${fading}`,
            }}
          >
            {isReady ? (
              <Image
                loading="lazy"
                src={`${thumbnailUrl}`}
                sx={{
                  backgroundColor: "surface",
                  position: "absolute",
                  top: 0,
                  bottom: 0,
                  left: 0,
                  right: 0,
                  cursor: "pointer",
                }}
              />
            ) : (
              <Typography.Body
                sx={{
                  display: "flex",
                  position: "absolute",
                  top: 0,
                  bottom: 0,
                  left: 0,
                  right: 0,
                  alignItems: "center",
                  justifyContent: "center",
                  fontWeight: "bold",
                  color: "textDisabled",
                }}
              >
                Processing
              </Typography.Body>
            )}
          </Box>
        </Box>
      </Box>
      {videoDialog.visible && (
        <VideoPlayerDialog
          studentId={studentId}
          src={playbackUrl}
          onClose={videoDialog.hide}
          thumbnailUrl={originalThumbnailUrl}
          createdAt={createdAt}
          videoId={videoId}
        />
      )}
    </Fragment>
  )
}

export default PageGallery
