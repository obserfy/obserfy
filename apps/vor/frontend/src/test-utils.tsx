/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
