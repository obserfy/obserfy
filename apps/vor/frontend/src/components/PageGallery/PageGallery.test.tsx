import { render } from "../../test-utils"
import PageGallery from "./PageGallery"

describe("PageGallery", () => {
  it("should render correctly", () => {
    const { container } = render(<PageGallery studentId="asdfasd" />)
    expect(container).toMatchSnapshot()
  })
})
