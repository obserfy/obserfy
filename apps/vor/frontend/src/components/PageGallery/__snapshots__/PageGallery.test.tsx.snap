// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageGallery should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surfaceBlurNonTransparent);
  z-index: 10;
  position: -webkit-sticky;
  position: sticky;
  top: -48px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-5:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-5:hover,
.emotion-5:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
}

.emotion-8 {
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-8:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-8:last-child>.splitter {
  display: none;
}

.emotion-9:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-10 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-14:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-14:last-child>.splitter {
  display: none;
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  overflow-x: auto;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 100%;
  width: 100%;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  padding-left: 4px;
  padding-right: 4px;
  margin-bottom: 0;
  color: var(--theme-ui-colors-textPrimary);
  white-space: nowrap;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 12px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 10;
  font-size: 14px;
  margin-left: 16px;
  margin-right: 16px;
}

.emotion-17:after {
  background-color: var(--theme-ui-colors-textPrimary);
  border-radius: 2px 2px 0 0;
  width: 125%;
  height: 3px;
  content: '';
  margin-top: 8px;
  -webkit-transition: width 100ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: width 100ms cubic-bezier(0.0, 0.0, 0.2, 1);
}

@media screen and (min-width: 40em) {
  .emotion-17 {
    font-size: 14px;
  }
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  padding-left: 4px;
  padding-right: 4px;
  margin-bottom: 0;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 12px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 10;
  font-size: 14px;
  margin-left: 16px;
  margin-right: 16px;
}

.emotion-18:after {
  background-color: var(--theme-ui-colors-textPrimary);
  border-radius: 2px 2px 0 0;
  width: 0%;
  height: 3px;
  content: '';
  margin-top: 8px;
  -webkit-transition: width 100ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: width 100ms cubic-bezier(0.0, 0.0, 0.2, 1);
}

@media screen and (min-width: 40em) {
  .emotion-18 {
    font-size: 14px;
  }
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  height: 1px;
  width: 100%;
  background-color: var(--theme-ui-colors-border);
  position: absolute;
  bottom: 0;
}

.emotion-21 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
  margin-left: 0;
  width: 100%;
  display: inline-block;
}

@media screen and (min-width: 40em) {
  .emotion-22 {
    padding-bottom: 8px;
    margin-left: auto;
  }
}

@media screen and (min-width: 40em) {
  .emotion-22 {
    width: auto;
  }
}

.emotion-23 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  width: 100%;
}

.emotion-23:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-23:hover,
.emotion-23:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-23:hover:disabled,
.emotion-23:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

@media screen and (min-width: 40em) {
  .emotion-23 {
    width: auto;
  }
}

.emotion-24 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-right: 8px;
  fill: var(--theme-ui-colors-onPrimary);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-25 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 8px;
  padding-right: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media screen and (min-width: 40em) {
  .emotion-25 {
    padding-left: 8px;
    padding-right: 8px;
  }
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <a
            class="emotion-4"
            data-cy="back-button"
            href="/dashboard/students/overview?studentId=asdfasd"
          >
            <button
              class="emotion-5"
            >
              <div
                class="emotion-6"
              />
            </button>
          </a>
          <div
            class="emotion-7"
          >
            <a
              class="emotion-8"
              href="/dashboard/students"
            >
              <span
                class="emotion-9"
              >
                Students
              </span>
              <span
                class="splitter emotion-10"
              >
                /
              </span>
            </a>
            <a
              class="emotion-8"
              href="/dashboard/students/overview?studentId=asdfasd"
            >
              <span
                class="emotion-9"
              />
              <span
                class="splitter emotion-10"
              >
                /
              </span>
            </a>
            <div
              class="emotion-14"
            >
              Gallery
              <span
                class="splitter"
              >
                /
              </span>
            </div>
          </div>
        </div>
        <div
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-17"
            >
              Images
            </p>
            <p
              class="emotion-18"
            >
              Videos
            </p>
          </div>
          <div
            class="emotion-19"
          />
        </div>
      </div>
      <div
        class="emotion-1"
      >
        <div
          class="emotion-21"
        >
          <label
            class="emotion-22"
          >
            <input
              accept="image/*"
              data-cy="upload-image"
              style="display: none;"
              type="file"
            />
            <div
              class="emotion-23"
            >
              <div
                class="emotion-24"
              />
              Upload Photo
            </div>
          </label>
        </div>
        <div
          class="emotion-25"
        />
      </div>
    </div>
  </div>
</div>
`;
